shopfloor/services/zone_picking.py

```diff
@@ -877,15 +877,16 @@ def _set_destination_location(self, move_line, quantity, confirmation, location)
             )
             return (location_changed, response)
 
-        if not confirmation and self.is_dest_location_to_confirm(
+        if confirmation != location.barcode and self.is_dest_location_to_confirm(
             move_line.location_dest_id, location
         ):
             response = self._response_for_set_line_destination(
                 move_line,
                 message=self.msg_store.confirm_location_changed(
                     move_line.location_dest_id, location
                 ),
-                confirmation_required=True,
+                # TODO fix this
+                confirmation_required=location.barcode,
```

Comment:
The scanned barcode must be in the method arguments and used in the comparisons

Reply:
This function is called when the barcode was used and a location matched the barcode.
The todo is not relevant any more

Reply:
Removed

Reply:
The location is matched from the scanned barcode by parsing it. Maybe the barcode is the location barcode but maybe not, you don't know what the parser is doing.

Reply:
If `_set_destination_location` is called it is because the barcode scanned does match a location.
So the barcode scanned is equal to the location barcode.

Reply:
No.
If the barcode is the location barcode (A), then it will match a location (B).
However, it's not because A implies B that B implies A. This is wrong.